Upstream: I honestly don't know where to send it...

From 14b5a8af6d97b25ce8255cf2420d1e88f88cc48d Mon Sep 17 00:00:00 2001
From: Heiko Becker <heirecka@exherbo.org>
Date: Mon, 7 Jan 2019 18:05:33 +0100
Subject: [PATCH] Fix build with glibc 2.28

From [1]: "The macros 'major', 'minor', and 'makedev' are now only
available from the header <sys/sysmacros.h>; not from <sys/types.h>
...<sys/sysmacros.h> is a GNU extension.  Portable programs that
require these macros should first include <sys/types.h>, and then
include <sys/sysmacros.h> if __GNU_LIBRARY__ is defined."

[1] https://www.sourceware.org/ml/libc-alpha/2018-08/msg00003.html
---
 growisofs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/growisofs.c b/growisofs.c
index 6c47192..4b9513e 100644
--- a/growisofs.c
+++ b/growisofs.c
@@ -444,6 +444,10 @@
 #include <assert.h>
 #include "mp.h"
 
+#if defined(__GNU_LIBRARY__)
+# include <sys/sysmacros.h>
+#endif
+
 #if defined(__unix) || defined(__unix__)
 # include <unistd.h>
 # include <sys/time.h>
-- 
2.20.1

